bool finish
---
int8 select_motion